### PR TITLE
Support for single node chains.

### PR DESCRIPTION
**Pull request by kape1395**
DKG and Chain adjusted to work with N=1 as well.

### CODE REVIEW

File: packages/dkg/proc.go [hunk 1]
```diff
@@ -178,6 +195,9 @@ func (p *proc) processLoop(timeout time.Duration, doneCh chan map[uint16]*peerin
 //
 func (p *proc) rabinStep1R21SendDealsMakeSent(step byte, initRecv *peering.RecvEvent, prevMsgs map[uint16]*peering.PeerMessage) (map[uint16]*peering.PeerMessage, error) {
 	var err error
+	if p.dkgImpl == nil {
+		return nil, errors.New("unexpected_step_for_n=1")
```
Comment (dessaya) [inline packages/dkg/proc.go]: Just curious, why use _ instead of spaces here?

**Review by dessaya — APPROVED** [review]

File: packages/dkg/proc.go (hunk 1 above)
Comment (kape1395) [reply packages/dkg/proc.go]: I'm not sure either. The style guide says it should begin with lower case and end without a period. Intuitively I assume, if that's not a sentence, then maybe a mnemonic. I can change it to spaces, if that's more common.

Comment (lunfardo314) [reply packages/dkg/proc.go]: I'd use spaces but it is a matter of taste I think

**Review by lunfardo314 — APPROVED** [review]

File: packages/dkg/proc.go (hunk 1 above)
Comment (kape1395) [reply packages/dkg/proc.go]: I will fix that.

**Pull request merged**